lib/engine/game/g_18_sj.rb

```diff
@@ -680,11 +680,11 @@ def gkb_bonus(stops)
 
         return bonus unless @gkb_bonus.positive?
 
-        revenue = 0
-        stops.select { |s| GKB_HEXES.include?(s.hex.name) }.each { |_s| revenue += @gkb_bonus }
-        if revenue.positive?
-          bonus[:revenue] = revenue
+        bonuses = stops.count { |s| GKB_HEXES.include?(s.hex.name) }
+        if bonuses.positive?
+          bonus[:revenue] = @gkb_bonus * bonuses
           bonus[:description] = 'GKB'
+          bonus[:description] += "x#{bonuses}" if bonuses > 1
```

Comment:
in here do route.clear_cache or something

Reply:
actually no, how exactly does the ability work and where is it applied?

Reply:
@tobymao As it is now the ability makes a button appear when you have the run route view up, and there are attempts left. If you click on the button a choose action is triggered which change the route revenue and string calculation change in the 18SJ game file. (A state is set in 18_sj/game.rb which is used when calculating route revenue.)
This does work if you in any way updates the route(s) in the GUI, but not for an existing route.

Reply:
@tobymao Have now changed so that ability is used instead. Have tried this with the game in question and it works fine, decrease the amount for each try, and remove the ability when used up.